test/test_common/simulated_time_system_test.cc

```diff
@@ -210,13 +233,26 @@ TEST_F(SimulatedTimeSystemTest, TimerOrderDisableAndRescheduleTimer) {
   // because it is scheduled with zero delay. Timer 2 executes in a later iteration because it is
   // re-enabled with a non-zero timeout.
   advanceMsAndLoop(5);
-  EXPECT_EQ("p0314", output_);
+  if (activateTimersNextEventLoop()) {
+#ifdef WIN32
+    // The event loop runs for a single iteration in NonBlock mode on Windows. Force it to run again
```

Comment:
should we try to get sim-time to behave the same in win32 as it does in linux in the class itself rather than ifdefing in the unit test?

Reply:
Do you have a suggestion on how to accomplish that?

The problem is that on Windows calls to
  base_scheduler_.run(Dispatcher::RunType::NonBlock);
do not run to completion in cases where timers are ready before polling for fd events because of the single iteration of the event loop hack they have in place because otherwise the use of LEVEL mode for fds results in the call to 'run' going into an infinite loop.  The time system does not override the base scheduler so there's no room for me to add a level of hacks to make Windows behavior nicer.  Some changes to libevent that add some mechanism to specify the maximum number of iterations before exit may allow me to improve this code, but I doubt those changes would make sense in a greater context.

Reply:
I don't. My sense is that if this behaves differently for Windows than non-Windows that other tests will run into trouble as well besides this one.

But clearly the direction you are going is fine too, which is to make everything as non-deterministic as possible, and force tests to clean up their flakes.

But maybe the text you wrote in this PR comment would be appropriate to add to the comment here, just so readers have more context.

Reply:
There is no way to make windows and non-windows operate consistently.  One thing I can do for windows is run callbacks in the same iteration to get slightly more consistent behavior, but there will still be observable changes that will require other forms of #ifdefs in this specific test.  Neither option sounds any good.

Reply:
per above I'm good to go with what you have, and I made a suggestion to include the more detailed comment you made about why the behavior is different.

I am OK with you adding that in or not, and I'll leave it up to you.

Reply:
Done.